File: JavaPractice/InsertionSort.java
public class InsertionSort{
	public static void main(String[] args){
		int []arr = new int[] {4,3,2,1,0,0,3};
		int temp = 0;
		for(int i=1;i<arr.length;i++){
			int j=i;
			while(j>0 && arr[j-1]>arr[j]){
				temp = arr[j];
				arr[j] = arr[j-1];
				arr[j-1] = temp;
				j--;
			}
		}

		for(int k=0;k<arr.length;k++){
			System.out.print(arr[k]+" ");
		}
	}
}

File: Sorting/InsertionSort/insertionSort.cpp
// Insertion Sort - O(n2)

void insertionSort(int arr[], int n)
{
    for(int i=1;i<n;i++)
    {
        int j=i-1;
        int currentElement = arr[i];
        while(j>=0 && currentElement < arr[j])
        {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = currentElement;
    }
}